src/js/classes/resizable.js

```diff
@@ -0,0 +1,503 @@
+        var _change = {
+          e: function(dx) {
+            return { width: originalSize.width + dx };
+          },
+          w: function(dx) {
+            var cs = originalSize, sp = originalPosition;
+            return { left: sp.left + dx, width: cs.width - dx };
+          },
+          n: function(dx, dy) {
+            var cs = originalSize, sp = originalPosition;
+            return { top: sp.top + dy, height: cs.height - dy };
+          },
+          s: function(dx, dy) {
+            return { height: originalSize.height + dy };
+          },
+          se: function(dx, dy) {
+            return angular.extend(this.s.apply(this, arguments),
+              this.e.apply(this, [ dx, dy ]));
+          },
+          sw: function(dx, dy) {
+            return angular.extend(this.s.apply(this, arguments),
+              this.w.apply(this, [ dx, dy ]));
+          },
+          ne: function(dx, dy) {
+            return angular.extend(this.n.apply(this, arguments),
+              this.e.apply(this, [ dx, dy ]));
+          },
+          nw: function(dx, dy) {
+            return angular.extend(this.n.apply(this, arguments),
+              this.w.apply(this, [ dx, dy ]));
+          }
+        };
+
+        return _change[this._axis](dx, dy);
+      };
+
+      /**
+       * Updates incoming data to fit ratio.
+       *
+       * @param data - boundaries data. 
+       */
+      ResizableFactory.prototype.updateRatio = function(data) {
+        if (angular.isNumber(data.height)) {
+          data.width = (data.height * this._ratio);
+        } else if (angular.isNumber(data.width)) {
+          data.height = (data.width / this._ratio);
+        }
+
+        if (this._axis === 'sw') {
+          data.left = this._position.left + (this._size.width - data.width);
+          data.top = null;
+        }
+        if (this._axis === 'nw') {
+          data.top = this._position.top + (this._size.height - data.height);
+          data.left = this._position.left + (this._size.width - data.width);
+        }
+
+        return data;
+      };
+
+
+      /**
+       * Updates current position and size from boundaries data.
+       *
+       * @param data - boundaries data. 
+       */
+      ResizableFactory.prototype.updateSizeAndPosition = function(data) {
+        if (angular.isNumber(data.left)) {
+          this._position.left = data.left;
+        }
+        if (angular.isNumber(data.top)) {
+          this._position.top = data.top;
+        }
+        if (angular.isNumber(data.height)) {
+          this._size.height = data.height;
+        }
+        if (angular.isNumber(data.width)) {
+          this._size.width = data.width;
+        }
+      };
+
+      /**
+       * Calculates new virtual boundaries values depends on options and ratio.
+       */
+      ResizableFactory.prototype.updateVirtualBoundaries = function() {
+        var pMinWidth, pMaxWidth, pMinHeight, pMaxHeight, b,
+            o = this._options;
+
+        b = {

```

Comment:
It is actually a shorteners. Function is not big. so defining `o` at the top makes code more readable.